src/StelLogger.cpp

```diff
@@ -44,193 +78,352 @@ void StelLogger::init(const QString& logFilePath)
 
-		if(systemLines.at(i).contains("Processor", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed().replace("Unknown", "Apple M1"));
+	int ncpu = sysconf( _SC_NPROCESSORS_ONLN );
+	writeLog(QString("Processor logical cores: %1").arg(ncpu));
 
-		if(systemLines.at(i).contains("Memory", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed());
+	// memory info
+	int64_t totalRAM = (size_t)sysconf( _SC_PHYS_PAGES ) * (size_t)sysconf( _SC_PAGESIZE );
+	writeLog(QString("Total physical memory: %1 MB").arg(totalRAM/(1024<<10)));
+#endif
 
-		// graphics/display overview
-		if(systemLines.at(i).contains("Resolution", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed());
+#ifdef Q_OS_HAIKU
+	// Idea and some code is catching from Haiku OS (MIT license)
+	uint32 topologyNodeCount = 0;
+	cpu_topology_node_info* topology = NULL;
 
-		if(systemLines.at(i).contains("VRAM", Qt::CaseInsensitive))
-			writeLog(systemLines.at(i).trimmed());
-	}
+	get_cpu_topology_info(NULL, &topologyNodeCount);
+	if (topologyNodeCount != 0)
+		topology = new cpu_topology_node_info[topologyNodeCount];
+	get_cpu_topology_info(topology, &topologyNodeCount);
 
-#elif defined Q_OS_BSD4
-	QProcess dmesg;
-	dmesg.start("/sbin/dmesg", {}, QIODevice::ReadOnly);
-	dmesg.waitForStarted();
-	dmesg.waitForFinished();
-	const QString dmesgData(dmesg.readAll());
-	QStringList dmesgLines = dmesgData.split('\n', Qt::SkipEmptyParts);
-	for (int i = 0; i<dmesgLines.size(); i++)
+	enum cpu_platform platform = B_CPU_UNKNOWN;
+	enum cpu_vendor cpuVendor = B_CPU_VENDOR_UNKNOWN;
+	uint32 cpuModel = 0;
+
+	for (uint32 i = 0; i < topologyNodeCount; i++)
 	{
-		if (dmesgLines.at(i).contains("memory"))
-		{
-			writeLog(dmesgLines.at(i).trimmed());
-		}
-		if (dmesgLines.at(i).contains("CPU"))
-		{
-			writeLog(dmesgLines.at(i).trimmed());
-		}
-		if (dmesgLines.at(i).contains("VGA"))
+		switch (topology[i].type)
 		{
-			writeLog(dmesgLines.at(i).trimmed());
+			case B_TOPOLOGY_ROOT:
+				platform = topology[i].data.root.platform;
+				break;
+			case B_TOPOLOGY_PACKAGE:
+				cpuVendor = topology[i].data.package.vendor;
+				break;
+			case B_TOPOLOGY_CORE:
+				cpuModel = topology[i].data.core.model;
+				break;
+			default:
+				break;
 		}
 	}
+	delete[] topology;
+
+	int32 frequency = get_rounded_cpu_speed();
+	QString clockSpeed;
+	if (frequency < 1000)
+		clockSpeed = QString("%1 MHz").arg(frequency);
+	else
+		clockSpeed = QString("%1 GHz").arg(QString::number(frequency/1000.f, 'f', 2));
+
+	writeLog(QString("Processor name: %1 %2 @ %3").arg(get_cpu_vendor_string(cpuVendor), get_cpu_model_string(platform, cpuVendor, cpuModel), clockSpeed));
+	writeLog(QString("Processor speed: %1 MHz").arg(frequency));
+
+	system_info hwinfo;
+	get_system_info(&hwinfo);
+
+	writeLog(QString("Processor logical cores: %1").arg(hwinfo.cpu_count));
 
+	// memory info
+	int totalRAM = round(hwinfo.max_pages*B_PAGE_SIZE/1048576.0 + hwinfo.ignored_pages*B_PAGE_SIZE/1048576.0);
```

Comment:
`int` is not future-proof. In principle a 64-bit system may have up to 2<sup>64</sup> bytes of RAM, which is 2<sup>44</sup> megabytes—not representable in a 32-bit `int`.